wagtailvideos/models.py

```diff
@@ -233,6 +236,23 @@ def get_transcode(self, media_format):
         except Transcode.DoesNotExist:
             return self.do_transcode(media_format)
 
+    def video_tag(self, attrs):
+        kw_attrs = {}
+
+        if self.thumbnail:
+            kw_attrs['poster'] = self.thumbnail.url
+
+        mime = mimetypes.MimeTypes()
+        sources = ["<source src='{0}' type='{1}'>"
+                   .format(self.url, mime.guess_type(self.url)[0])]
+
+        transcodes = self.transcodes.exclude(processing=True).filter(error_message__exact='')
+        for transcode in transcodes:
+            sources.append("<source src='{0}' type='video/{1}' >".format(transcode.url, transcode.media_format.name))
+        sources.append("<p>Sorry, your browser doesn't support playback for this video</p>")
+        return mark_safe(
+            "<video {0} {1}>{2}</video>".format(attrs, flatatt(kw_attrs), "\n".join(sources)))
```

Comment:
`attrs` should be passed through `flatatt` as well, so a dict of attrs can be passed in. Also, `flatatt` adds a space before each attr, so you can use it like:
```python
"<video{0}{1}>{2}</video>.format(flatatt(attrs), flatatt(kw_attrs), ...)
```